File: win-prog/ProgWinCS/Buttons and Labels and Scrolls (Oh, My!)/ButtonStyles/ButtonStyles.cs
//-------------------------------------------
// ButtonStyles.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ButtonStyles: Form
{
     public static void Main()
     {
          Application.Run(new ButtonStyles());
     }
     public ButtonStyles()
     {
          Text = "Button Styles";

          int y = 0;

          foreach (FlatStyle fs in Enum.GetValues(typeof(FlatStyle)))
          {
               Button btn    = new Button();
               btn.Parent    = this;
               btn.FlatStyle = fs;
               btn.Text      = fs.ToString();
               btn.Location  = new Point(50, y += 50);
          }
     }
}

File: win-prog/ProgWinVB/Buttons and Labels and Scrolls (Oh, My!)/ButtonStyles/ButtonStyles.vb
'---------------------------------------------
' ButtonStyles.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ButtonStyles
    Inherits Form

    Shared Sub Main()
        Application.Run(New ButtonStyles())
    End Sub

    Sub New()
        Text = "Button Styles"

        Dim y As Integer = 50
        Dim fs As FlatStyle

        For Each fs In System.Enum.GetValues(GetType(FlatStyle))
            Dim btn As New Button()
            btn.Parent = Me
            btn.FlatStyle = fs
            btn.Text = fs.ToString()
            btn.Location = New Point(50, y)
            y += 50
        Next fs
    End Sub
End Class
